### PR TITLE
Add OpenSSL::SSL::Context#cipher_suites= and OpenSSL::SSL::Context::CIPHER_SUITES_* constants

### PR DESCRIPTION
**Pull request by bcardiff**
On OpenSSL 1.1.1g setting ciphers to the modern constant does not work.

```crystal
require "openssl"
ctx = OpenSSL::SSL::Context::Server.new
ctx.ciphers = OpenSSL::SSL::Context::CIPHERS_MODERN
# => Unhandled exception: SSL_CTX_set_cipher_list: error:1410D0B9:SSL routines:SSL_CTX_set_cipher_list:no cipher match (OpenSSL::Error)
```

The ciphers listed in CIPHERS_MODERN should be set via `SSL_CTX_set_ciphersuites` and not via `SSL_CTX_set_cipher_list`.

* https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_ciphersuites.html
* https://www.openssl.org/blog/blog/2017/05/04/tlsv1.3/

This PR add the required binding and constants.

The cipher separator was changed to `:` because although 
> The cipher list consists of one or more cipher strings separated by colons. Commas or spaces are also acceptable separators **but colons are normally used**.
> Ref: https://www.openssl.org/docs/man1.0.2/man1/ciphers.html

For ciphersuites only colons are allowed as separators, so it's better to use a separator that works for both.

cc: @carlhoerberg @straight-shoota

### CODE REVIEW

File: scripts/generate_ssl_server_defaults.cr [hunk 1]
```diff
@@ -24,7 +24,7 @@ File.open(DEFAULTS_FILE, "w") do |file|
     clients = configuration["oldest_clients"].as_a
     ciphersuites = configuration["ciphersuites"].as_a
     ciphers = configuration["ciphers"]["openssl"].as_a
-    disabled_ciphers = %w(!RC4 !aNULL !eNULL !LOW !3DES !MD5 !EXP !PSK !SRP !DSS)
+    disabled_ciphers = %w(!RC4:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!PSK:!SRP:!DSS)
```
Comment (asterite) [inline scripts/generate_ssl_server_defaults.cr:27]: This is not correct. `%w` is an array literal.

**Comment by jhass** [comment]
Shouldn't this update 

https://github.com/crystal-lang/crystal/blob/8c182b8ab4445f84d08022546b400c7aa3d4d442/src/openssl/ssl/context.cr#L131

and 

https://github.com/crystal-lang/crystal/blob/8c182b8ab4445f84d08022546b400c7aa3d4d442/src/openssl/ssl/context.cr#L49

too?

**Comment by bcardiff** [comment]
Regarding the abstraction, the further we go from openssl, the harder it will be track down what is needed IMO.

Regarding the default values, I am not sure what is preferred. The current policy is that client use old for compatibility and servers intermediate for security. This was stated in #9459. I guess the modern was too restrictive for the server. @straight-shoota do you recall any other criteria?

**Comment by jhass** [comment]
Right, I somehow overlooked that it only doesn't work for `MODERN`. 

But, OpenSSL's API is known for its foot guns, this seems like yet another one. Abstracting some away doesn't seem like a bad thing to me. 

Maybe instead of the `context.ciphers =  SOME_CONSTANT` API we should move to something like `context.set_modern_ciphers` and make the constants internal. Then those methods could do the right thing depending on the OpenSSL version.

It just feels wrong to force each consumer to do the version check themselves :/

**Comment by bcardiff** [comment]
@jhass I implemented set_modern_ciphers and others. I think is a good idea to have those methods and abstract the version discrepancies a bit.

**Review by jhass — APPROVED** [review]

**Review by straight-shoota — COMMENTED** [review]
:+1:

**Comment by asterite** [comment]
@straight-shoota Welcome back! How have you been? 👋

**Comment by straight-shoota** [comment]
I'm doing fine, thanks. Just didn't get around here lately. I guess it's good to get some distance from time to time.

**Comment by asterite** [comment]
Definitely! I do that almost every month, or even week, haha, depending on how I'm feeling.

**Pull request merged**